test/unit/conditions/context.test.ts

```diff
@@ -90,20 +90,16 @@ describe('context parameters', () => {
     };
     const condition = new ContractCondition(conditionObj);
     const conditionExpr = new ConditionExpression(condition);
-    const conditionContext = conditionExpr.buildContext(provider, {}, signer);
     expect(conditionExpr.contextRequiresSigner()).toBe(true);
```

Comment:
I think it's good to test the building works i.e.
```suggestion
    expect(conditionExpr.contextRequiresSigner()).toBe(true);

    // check that conditionContext created successfully
    expect(conditionExpr.buildContext(provider, {}, signer)...to succeed
    
    // try to create condition context without signer
    expect(conditionExpr.buildContext(provider, {})....to fail
```